yb-voyager/cmd/importDataRandomFileBatchProducer_test.go

```diff
@@ -0,0 +1,1254 @@
+}
+
+func TestRandomBatchProducer_Resumption_PartialBatchesProduced_NoneConsumed(t *testing.T) {
+	// Create a file with 10 batches (20 rows, 2 rows per batch)
+	ldataDir, lexportDir, state, errorHandler, progressReporter, err := setupExportDirAndImportDependencies(2, 1024)
+	require.NoError(t, err)
+
+	if ldataDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", ldataDir))
+	}
+	if lexportDir != "" {
+		defer os.RemoveAll(fmt.Sprintf("%s/", lexportDir))
+	}
+
+	// Create file with 20 data rows (10 batches of 2 rows each)
+	fileContents := "id,val\n"
+	for i := 1; i <= 20; i++ {
+		fileContents += fmt.Sprintf("%d,value%d\n", i, i)
+	}
+
+	_, task, err := createFileAndTask(lexportDir, fileContents, ldataDir, "test_table", 1)
+	require.NoError(t, err)
+
+	// Create sequential producer
+	sequentialProducer, err := NewSequentialFileBatchProducer(task, state, errorHandler, progressReporter)
+	require.NoError(t, err)
+
+	// Wrap it with a delayed version that sleeps 100ms per batch
+	// This ensures we have time to observe partial batches
+	delayedSequential := &delayedSequentialFileBatchProducer{
+		SequentialFileBatchProducer: sequentialProducer,
+		sleepDuration:               500 * time.Millisecond,
+	}
+
+	// Create random producer using the delayed sequential producer
+	producer := newRandomFileBatchProducer(delayedSequential, task)
+
+	// Wait for some batches to be produced (but not all)
+	// We want to ensure we have partial batches (more than 0, less than 10)
+	// Don't consume any batches yet - we want to test resumption with batches in memory
+	var batchesProduced int
+	deadline := time.Now().Add(5 * time.Second)
+	for time.Now().Before(deadline) {
+		producer.mu.Lock()
+		batchesProduced = len(producer.sequentiallyProducedBatches)
+		producer.mu.Unlock()
+		// We want at least 3 batches but not all 10
+		// keeping it to below 7 because this is potential
+		// race condition between here and when we close the producer.
+		if batchesProduced >= 3 && batchesProduced < 7 {
+			// do not unlock because we don't want producer to add more batches.
+			// we are going to close the producer anyway.
+			break
+		}
+
+		time.Sleep(50 * time.Millisecond)
+	}
+
+	// Verify we have partial batches
+	require.GreaterOrEqual(t, batchesProduced, 3, "Should have at least 3 batches produced")
+	require.Less(t, batchesProduced, 7, "Should have less than 10 batches produced (partial)")
+
+	// Note: The batches produced are saved to state(disk) by SequentialFileBatchProducer.
+	// When we close, the batches in memory (sequentiallyProducedBatches) are lost,
+	// but the batches already saved to state(disk) will be recovered on resumption.
+
+	// Close the producer to simulate interruption
+	producer.Close()
+
+	// Create a new producer (simulating resumption)
+	// This should recover the batches that were already produced
+	sequentialProducer2, err := NewSequentialFileBatchProducer(task, state, errorHandler, progressReporter)
+	require.NoError(t, err)
+
```

Comment:
Maybe before starting the random batch producer, we can verify the batches produced before starting the rest of the production?